src/file/FileDialog.tsx

```diff
@@ -215,6 +218,74 @@ const OpenLocalFile: React.FC<SourceTabProps> = ({ onDismiss }) => {
     );
 };
 
+const ImportFromString: React.FC<SourceTabProps> = ({ onDismiss }) => {
+    const loadScene = useLoadScene();
+    const setSavedState = useSetSavedState();
+    const isDirty = useIsDirty();
+    const theme = useTheme();
+    const [data, setData] = useState<string | undefined>('');
+    const [error, setError] = useState<string | undefined>('');
+
+    const importCallback = useCallback(async () => {
+        if (!data) {
+            return;
+        }
+
+        if (isDirty) {
+            if (!(await confirmUnsavedChanges(theme))) {
+                return;
+            }
+        }
+
+        let scene;
+        try {
+            scene = await textToScene(data);
+        } catch (ex) {
+            console.log(ex);
+            setError('Invalid Import String');
```

Comment:
```suggestion
            setError('Invalid import string');
```

Super minor: since this isn't a header/title, sentence case (only first word capitalized) would be more appropriate.

I'm also not sure how long or meaningful the exception messages typically would be here, but if they are reasonably short and understandable, something like this could be helpful:

```ts
            setError(`Invalid import string. ${ex}`);
```